packages/cli/internal/pkg/aws/cwl/get_logs.go

```diff
@@ -51,9 +51,22 @@ func (c Client) GetLogsPaginated(input GetLogsInput) LogPaginator {
 }
 
 func formatEvents(events []types.FilteredLogEvent) []string {
-	logs := make([]string, len(events))
-	for i, event := range events {
-		logs[i] = formatEvent(event)
+	logsByStream := make(map[string][]*types.FilteredLogEvent)
+	for index, _ := range events {
+		event := events[index]
+		logsByStream[*event.LogStreamName] = append(logsByStream[*event.LogStreamName], &event)
+	}
+
+	return convertStreamLogsToLogs(logsByStream, len(events))
+}
+
+func convertStreamLogsToLogs(logsByStream map[string][]*types.FilteredLogEvent, eventSize int) []string {
+	logs, index := make([]string, eventSize), 0
+	for _, eventList := range logsByStream {
```

Comment:
Although we are performing two loops, it requires the exact same number of loops as before and is effectively the equivalent of using a spread operator in Go (see here for more details: https://github.com/golang/go/blob/master/src/reflect/value.go#L2640 ). We can't use the spread operator as we only have a memory pointer and need to format the events so that they print correctly.